include/mata/nfa.hh

```diff
@@ -525,19 +610,44 @@ public:
         m_num_of_requested_states = 0;
     }
 
+    // this is exact equality of automata, including state numbering (so even stronger than isomorphism)
+    // essentially only useful for testing purposes
```

Comment:
Even for testing purposes I would consider ordering the tests in order of "least expensive" to "most expensive"